build.go

```diff
@@ -34,46 +34,139 @@ import (
-	ContainerConfig    ContainerConfig
+	// required. Name of output image.
+	Image string
+
+	// required. Builder image name.
+	Builder string
+
+	// URI to a buildpack registry. Used to
+	// add buildpacks to a build.
+	Registry string
+
+	// AppPath is the path to application bits.
+	// If unset it defaults to current working directory.
+	AppPath string
+
+	// Specify the run image the Image will be
+	// built atop.
+	RunImage string
+
+	// Used to determine a run-image mirror if Run Image is empty.
+	// Used in combination with Builder metadata to determine to the the 'best' mirror.
+	// 'best' is defined as:
+	//  - if Publish is true, the best mirror matches registry we are publishing to.
+	//  - if Publish is false, the best mirror matches a registry specified in Image.
+	//  - otherwise if both of the above did not match, use mirror specified in
+	//    the builder metadata
+	AdditionalMirrors map[string][]string
+
+	// User provided environment variables to the buildpacks.
+	// Buildpacks may both read and overwrite these values.
+	Env map[string]string
+
+	// Option passed directly to the lifecycle.
+	// If true, publishes Image directly to a registry.
+	// Assumes Image contains a valid registry with credentials
+	// provided by the docker client.
+	Publish bool
+
+	// Clear the build cache from previous builds.
+	ClearCache bool
+
+	// TrustBuilder when true optimizes builds by running
+	// all lifecycle phases in a single container.
+	// This places registry credentials on the builder's build image.
+	// Only trust builders from reputable sources.
+	TrustBuilder bool
+
+	// List of buildpack images or archives to add to a builder.
+	// These buildpacks may overwrite those on the builder if they
+	// share both an ID and Version with a buildpack on the builder.
+	Buildpacks []string
+
+	// Configure the proxy environment variables,
+	// These variables will only be set in the build image
+	// and will not be used if proxy env vars are already set.
+	ProxyConfig *ProxyConfig
+
+	// Configure network and volume mounts for the build containers.
+	ContainerConfig ContainerConfig
+
+	// Process type that will be used when setting container start command.
 	DefaultProcessType string
-	FileFilter         func(string) bool
-	PullPolicy         config.PullPolicy
+
+	// Filter files from the application source.
+	// If true include file, otherwise exclude.
+	FileFilter func(string) bool
+
+	// Strategy for updating local images before a build.
+	PullPolicy config.PullPolicy
 }
 
+// ProxyConfig specifies proxy setting to be set as environment variables in a container.
 type ProxyConfig struct {
-	HTTPProxy  string
-	HTTPSProxy string
-	NoProxy    string
+	HTTPProxy  string // Used to set HTTP_PROXY env var.
+	HTTPSProxy string // Used to set HTTPS_PROXY env var.
+	NoProxy    string // Used to set NO_PROXY env var.
 }
 
+// Additional configuration of the docker container that all build steps
+// occur within.
```

Comment:
```suggestion
// ContainerConfig is additional configuration of the docker container that all build steps
// occur within.
```

The GoDocs should all "start with the element being described".

> Doc comments work best as complete sentences, which allow a wide variety of automated presentations. The first sentence should be a one-sentence summary that starts with the name being declared.

Reference: https://golang.org/doc/effective_go.html